File: app/src/main/java/com/vwoom/timelapsecollection/viewmodels/DetailsViewModelFactory.java
package com.vwoom.timelapsecollection.viewmodels;

import android.util.Log;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.vwoom.timelapsecollection.database.TimeLapseDatabase;

public class DetailsViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final TimeLapseDatabase mTimeLapseDb;
    private final long mProjectId;

    private static final String TAG = DetailsViewModelFactory.class.getSimpleName();

    public DetailsViewModelFactory(TimeLapseDatabase timeLapseDatabase, long projectId){
        this.mTimeLapseDb = timeLapseDatabase;
        this.mProjectId = projectId;

        Log.d(TAG, "viewmodel factory firing up project id is " + projectId);
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass){
        return (T) new DetailsActivityViewModel(mTimeLapseDb, mProjectId);
    }

}

File: app/src/main/java/com/vwoom/timelapsegallery/details/DetailsViewModelFactory.kt
package com.vwoom.timelapsegallery.details

import android.util.Log
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.vwoom.timelapsegallery.data.TimeLapseDatabase

class DetailsViewModelFactory(private val mTimeLapseDb: TimeLapseDatabase, private val mProjectId: Long) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return DetailsActivityViewModel(mTimeLapseDb, mProjectId) as T
    }

    companion object {
        private val TAG = DetailsViewModelFactory::class.java.simpleName
    }

    init {
        Log.d(TAG, "viewmodel factory firing up project id is $mProjectId")
    }
}